test/keysym.c

```diff
@@ -391,7 +391,7 @@ main(void)
         test_icu_case_mappings(ks);
 #endif
     }
-    iter = xkb_keysym_iterator_unref(iter);
+    xkb_keysym_iterator_unref(iter);
```

Comment:
I remember you actually [asked me](https://github.com/xkbcommon/libxkbcommon/pull/414#discussion_r1421837962) 😉 to add this to avoid dangling pointers. I guess it does not matter here because we do not reuse the variable later.

Reply:
yeah. I'd personally prefer resetting it because it's a safety operation but tools like scan-build are useful so keeping them noise-free unfortunately trumps my personal preference :)

I'd probably leave the commits as-is but if you want to squash them please do so, I have no preference on that.